tools/cli/lib/api.c

```diff
@@ -159,47 +166,80 @@ TDNFCliListCommand(
     BAIL_ON_CLI_ERROR(dwError);
 
     dwError = pContext->pFnList(pContext, pListArgs, &pPkgInfo, &dwCount);
+    if (pCmdArgs->nJsonOutput && dwError == ERROR_TDNF_NO_MATCH)
+    {
+        dwError = 0;
+    }
     BAIL_ON_CLI_ERROR(dwError);
 
-    dwError = GetColumnWidths(COL_COUNT, nColPercents, nColWidths);
-    BAIL_ON_CLI_ERROR(dwError);
-
-    for(dwIndex = 0; dwIndex < dwCount; ++dwIndex)
+    if (pCmdArgs->nJsonOutput)
     {
-        pPkg = &pPkgInfo[dwIndex];
+        jd = jd_create(0);
+        CHECK_JD_NULL(jd);
+
+        jd_list_start(jd);
 
-        memset(szNameAndArch, 0, MAX_COL_LEN);
-        if(snprintf(
-            szNameAndArch,
-            MAX_COL_LEN,
-            "%s.%s",
-            pPkg->pszName,
-            pPkg->pszArch) < 0)
+    	for(dwIndex = 0; dwIndex < dwCount; ++dwIndex)
         {
-            dwError = errno;
-            BAIL_ON_CLI_ERROR(dwError);
+            jd_pkg = jd_create(0);
+            CHECK_JD_NULL(jd_pkg);
+
+            CHECK_JD_RC(jd_map_start(jd_pkg));
+            pPkg = &pPkgInfo[dwIndex];
+
+            CHECK_JD_RC(jd_map_add_string(jd_pkg, "Name", pPkg->pszName));
+            CHECK_JD_RC(jd_map_add_string(jd_pkg, "Arch", pPkg->pszArch));
+            CHECK_JD_RC(jd_map_add_fmt(jd_pkg, "Evr", "%s-%s", pPkg->pszVersion, pPkg->pszRelease));
+            CHECK_JD_RC(jd_map_add_string(jd_pkg, "Repo", pPkg->pszRepoName));
+
+            CHECK_JD_RC(jd_list_add_child(jd, jd_pkg));
+            jd_destroy(jd_pkg); jd_pkg = NULL;
         }
+        pr_json(jd->buf);
+        jd_destroy(jd); jd = NULL;
```

Comment:
new line please or use a macro.

Reply:
I like it this way. It's clear and doesn't waste space.